code/modules/antagonists/space_dragon/space_dragon.dm

```diff
@@ -47,6 +72,103 @@
+ *
+ * A simple update check which sees if we need to do anything based on the current state of the dragon's rifts.
+ *
+ */
+/datum/antagonist/space_dragon/proc/rift_checks()
+	if((rifts_charged == 3 || (SSshuttle.emergency.mode == SHUTTLE_DOCKED && rifts_charged > 0)) && !objective_complete)
+		victory()
+		return
+	if(riftTimer == -1)
+		return
+	riftTimer = min(riftTimer + 1, maxRiftTimer + 1)
+	if(riftTimer == (maxRiftTimer - 60))
+		to_chat(owner.current, span_boldwarning("You have a minute left to summon the rift! Get to it!"))
+		return
+	if(riftTimer >= maxRiftTimer)
+		to_chat(owner.current, span_boldwarning("You've failed to summon the rift in a timely manner! You're being pulled back from whence you came!"))
+		destroy_rifts()
+		SEND_SOUND(owner.current, sound('sound/magic/demon_dies.ogg'))
+		QDEL_NULL(owner.current)
+
+/**
+ * Destroys all of Space Dragon's current rifts.
+ *
+ * QDeletes all the current rifts after removing their references to other objects.
+ * Currently, the only reference they have is to the Dragon which created them, so we clear that before deleting them.
+ * Currently used when Space Dragon dies or one of his rifts is destroyed.
+ */
+/datum/antagonist/space_dragon/proc/destroy_rifts()
+	if(objective_complete)
+		return
+	rifts_charged = 0
+	owner.current.add_movespeed_modifier(/datum/movespeed_modifier/dragon_depression)
+	riftTimer = -1
+	SEND_SOUND(owner.current, sound('sound/vehicles/rocketlaunch.ogg'))
+	for(var/obj/structure/carp_rift/rift in rift_list)
+		rift.dragon = null
+		rift_list -= rift
+		if(!QDELETED(rift))
+			QDEL_NULL(rift)
+
+/**
+ * Sets up Space Dragon's victory for completing the objectives.
+ *
+ * Triggers when Space Dragon completes his objective.
+ * Calls the shuttle with a coefficient of 3, making it impossible to recall.
+ * Sets all of his rifts to allow for infinite sentient carp spawns
+ * Also plays appropiate sounds and CENTCOM messages.
+ */
+/datum/antagonist/space_dragon/proc/victory()
+	objective_complete = TRUE
+	permanant_empower()
+	var/datum/objective/summon_carp/main_objective = locate() in objectives
+	if(main_objective)
+		main_objective.completed = TRUE
+	priority_announce("A large amount of lifeforms have been detected approaching [station_name()] at extreme speeds. \
+	Remaining crew are advised to evacuate as soon as possible.", "Central Command Wildlife Observations")
+	sound_to_playing_players('sound/creatures/space_dragon_roar.ogg')
+	for(var/obj/structure/carp_rift/rift in rift_list)
+		rift.carp_stored = 999999
+		rift.time_charged = rift.max_charge
+
+/**
+ * Gives Space Dragon their the rift speed buff permanantly.
+ *
+ * Gives Space Dragon the enraged speed buff from charging rifts permanantly.
+ * Only happens in circumstances where Space Dragon completes their objective.
+ */
+/datum/antagonist/space_dragon/proc/permanant_empower()
+	owner.current.fully_heal()
```

Comment:
the comment above doesn't mention it fully healing you, is this something we want to keep? If so, we should add it to the comment.